File: swift/Sources/swift/exampleD.swift
import Foundation

class ShoppingBasketFactory {
 
    func getQuantity(of itemName: String) -> Int {
        fatalError("NotImplementedError: getQuantity function not implemented")
    }

    func calculateTotal() -> Decimal {
        fatalError("NotImplementedError: calculateTotal function not implemented")
    }
    
    static func fromJson(json: String) throws -> ShoppingBasket{
        fatalError("NotImplementedError: from_json function not implemented")
    }
}

File: typescript/src/exampleD.ts
export class ShoppingBasket {

  getQuantity(name: string): number {
    throw new Error("Not implemented yet");
  }

  calculateTotal(): number {
    throw new Error("Not implemented yet");
  }
}

export class ShoppingBasketFactory {
  static fromJson(json: string): ShoppingBasket {
    throw new Error("Not implemented yet");
  }
}
